openedx/adg/lms/applications/tests/test_helpers.py

```diff
@@ -324,3 +325,30 @@ def test_get_extra_context_for_application_review_page(mock_get_application_revi
     actual_context = get_extra_context_for_application_review_page(user_application)
 
     assert expected_context == actual_context
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize(
+    'is_staff, is_superuser, is_business_line_admin, expected_result',
+    [
+        (False, False, False, False),
+        (False, False, True, False),
+        (False, True, False, False),
+        (False, True, True, False),
+        (True, False, False, False),
+        (True, False, True, True),
+        (True, True, False, True),
+        (True, True, True, True),
+    ]
+)
+def test_has_admin_permissions(mocker, is_business_line_admin, is_superuser, is_staff, expected_result):
+    """
+    Test if the user is a superuser or an ADG admin or the admin of any Business line while having the staff
+    status
+    """
+    mocked_class = mocker.patch('openedx.adg.lms.applications.models.BusinessLine')
```

Comment:
Lets mock from where the function is used not from where it is created. [see](https://stackoverflow.com/a/32773203)

`openedx.adg.lms.applications.helpers.BusinessLine`

Reply:
This is not true in the case where the import is done inside a function.

If I use `openedx.adg.lms.applications.helpers.BusinessLine`, it results in an error.

Reply:
Please take a look at this [thread](https://stackoverflow.com/questions/22201583/patch-over-a-function-imported-inside-another-function).